src/o365/graph/commands/teams/teams-user-list.ts

```diff
@@ -0,0 +1,146 @@
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+import { GroupUser } from '../o365group/GroupUser';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  role?: string;
+}
+
+
+class TeamsUserListCommand extends GraphItemsListCommand<GroupUser> {
+  public get name(): string {
+    return `${commands.TEAMS_USER_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Lists Microsoft Teams teams users for a specified team';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    this.getOwners(cmd, args.options.teamId)
+      .then((): Promise<any> => {
+        if (args.options.role != "Owner") {
+
+          return this.getMembersAndGuests(cmd, args.options.teamId)
+            .then((): Promise<any> => {
+
+              // Filter out duplicate added values for owners (as they are returned as members as well)
+              this.items = this.items.filter((groupUser, index, self) =>
+                index === self.findIndex((t) => (
+                  t.id === groupUser.id && t.displayName === groupUser.displayName
+                ))
+              )
+
+              return Promise.resolve();
+            });
+        } else {
+          return Promise.resolve();
+        }
+      })
+
+      .then((): void => {
+        if (args.options.role) {
+          this.items = this.items.filter(i => i.userType === args.options.role)
+        }
+
+        cmd.log(this.items);
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  private getOwners(cmd: CommandInstance, teamId: string): Promise<void> {
+    let endpoint: string = `${auth.service.resource}/v1.0/groups/${teamId}/owners?$select=id,displayName,userPrincipalName,userType`;
+
+    return this.getAllItems(endpoint, cmd, true).then((): void => {
+
+      // Currently there is a bug in the Graph that returns Owners as userType 'member'
+      // We therefor update all returned user as owner  
+      for (var i in this.items) {
+        this.items[i].userType = "Owner";
+      }
+    });
+  }
+
+  private getMembersAndGuests(cmd: CommandInstance, teamId: string): Promise<any> {
+    let endpoint: string = `${auth.service.resource}/v1.0/groups/${teamId}/members?$select=id,displayName,userPrincipalName,userType`;
+    return this.getAllItems(endpoint, cmd, false);
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId <teamId>',
+        description: 'The GroupId of the team'
```

Comment:
In other commands we're referring to the team ID as `Team ID` so let's use it here instead.